src/test/java/api/loans/CheckOutByBarcodeTests.java

```diff
@@ -1916,6 +1930,127 @@ public void canOverrideManualAndAutomationPatronBlocksWhenBlocksArePresent() {
+    IndividualResource steve = usersFixture.steve(user ->
+      user.expires(DateTime.now().plusHours(1))
+        .active());
+
+    JsonObject response = checkOutFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(item)
+        .to(steve)
+        .at(CASE_YESTERDAY_OPEN_TODAY_CLOSED_TOMORROW_OPEN)).getJson();
+
+    assertThat(DateTime.parse(response.getString("dueDate")).toDateTime(),
+      is(YESTERDAY_OPEN_DAY.toDateTime(LocalTime.MIDNIGHT.minusSeconds(1), UTC)));
+  }
+
+  @Test
+  public void dueDateShouldBeTruncatedToThePatronsExpirationDateTimeIfKeepCurrentDueDateStrategy() {
+    use(buildLoanPolicyWithFixedLoan(KEEP_THE_CURRENT_DUE_DATE));
+
+    IndividualResource item = itemsFixture.basedUponNod();
+    DateTime patronExpirationDate = DateTime.now().plusDays(3).withZone(UTC);
+    IndividualResource steve = usersFixture.steve(user -> user.expires(patronExpirationDate));
+
+    JsonObject response = checkOutFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(item)
+        .to(steve)
+        .at(CASE_ONE_DAY_IS_OPEN_NEXT_TWO_DAYS_CLOSED)).getJson();
+
+    assertThat(DateTime.parse(response.getString("dueDate")).toDateTime(), is(patronExpirationDate));
+  }
+
+  @Test
+  public void dueDateShouldBeTruncatedToPatronsExpirationDateTimeIfKeepCurrentDueDateTimeStrategy() {
+    use(buildLoanPolicyWithRollingLoan(KEEP_THE_CURRENT_DUE_DATE_TIME));
+
+    IndividualResource item = itemsFixture.basedUponNod();
+    DateTime patronExpirationDate = DateTime.now().plusHours(1).withZone(UTC);
+    IndividualResource steve = usersFixture.steve(user -> user.expires(patronExpirationDate));
+
+    JsonObject response = checkOutFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(item)
+        .to(steve)
+        .at(CASE_ONE_DAY_IS_OPEN_NEXT_TWO_DAYS_CLOSED)).getJson();
+
+    assertThat(DateTime.parse(response.getString("dueDate")).toDateTime(), is(patronExpirationDate));
+  }
+
+  @Test
+  public void
+  dueDateShouldBeTruncatedToTheEndOfPreviousServicePointHoursIfMoveToTheEndOfCurrentHoursStrategy() {
+    use(buildLoanPolicyWithRollingLoan(MOVE_TO_END_OF_CURRENT_SERVICE_POINT_HOURS));
+
+    IndividualResource item = itemsFixture.basedUponNod();
+    DateTime patronExpirationDate = DateTime.now().plusHours(1).withZone(UTC);
+    IndividualResource steve = usersFixture.steve(user -> user.expires(patronExpirationDate));
+
+    JsonObject response = checkOutFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(item)
+        .to(steve)
+        .at(CASE_YESTERDAY_OPEN_TODAY_CLOSED_TOMORROW_OPEN)).getJson();
+
+    assertThat(DateTime.parse(response.getString("dueDate")).toDateTime(),
+      is(YESTERDAY_OPEN_DAY.toDateTime(END_TIME_SECOND_PERIOD, UTC)));
+  }
+
+  @Test
+  public void
+  dueDateShouldBeTruncatedToTheEndOfPreviousServicePointHoursIfMoveToTheBeginningOfNextStrategy() {
```

Comment:
Make sure it doesn't fail later in the day, when there is no "next open hours" period